### PR TITLE
(PE-33424) Add Snyk scan

### PR DESCRIPTION
**Pull request by rileynewton**

### CODE REVIEW

File: .github/workflows/synk-scan.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+    - name: Run Vanagon Snyk Scan - push
+      uses: puppetlabs/security-snyk-vanagon-action@v2.2.0
+      if: github.event_name == 'push'
+      with:
+        snykToken: ${{ secrets.SNYK_PE_TOKEN }}
+        snykOrg: 'puppet-foss'
```
Comment (rileynewton) [inline .github/workflows/synk-scan.yml:25]: Copied snykOrg from Lucy's #174 PR, not sure if it matters what we put here.

File: .github/workflows/snyk-scan.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+name: snyk vanagon check
+
+on:
+  push:
+  pull_request_target:
+      types: [opened, reopened, edited, synchronize]
```
Comment (donoghuc) [inline .github/workflows/snyk-scan.yml:7]: I'm kind of surprised this works... I would have thought we need:
```yaml
on:
  - push
  - pull_request_target:
      types: [opened, reopened, edited, synchronize]
```

What you have evaluates to this:
```
irb(main):004:0> YAML.load(File.read('foo.yaml'))
=> {true=>{"push"=>nil, "pull_request_target"=>{"types"=>["opened", "reopened", "edited", "synchronize"]}}}
```
Compare that to what i've suggested:
```
irb(main):003:0> YAML.load(File.read('foo.yaml'))
=> {true=>["push", {"pull_request_target"=>{"types"=>["opened", "reopened", "edited", "synchronize"]}}]}
```

Comment (rileynewton) [reply .github/workflows/snyk-scan.yml:7]: 👍  I was actually trying that syntax but github's linter rejects it probably mistakenly, will push that.

Comment (rileynewton) [reply .github/workflows/snyk-scan.yml]: It fails 😮  https://github.com/puppetlabs/bolt-vanagon/actions/runs/1764029008

Comment (donoghuc) [reply .github/workflows/snyk-scan.yml]: hmmmm

Comment (donoghuc) [reply .github/workflows/snyk-scan.yml]: Well if it is working then i guess its ok the way it was. yaml is hard

Comment (rileynewton) [reply .github/workflows/snyk-scan.yml:7]: Github has some very slippery syntax, thought there was a fix but that also didn't work https://github.community/t/multiple-event-triggers-with-config/16408/4 - oh well 🤷

**Review by donoghuc — APPROVED** [review]

File: .github/workflows/snyk-scan.yml (hunk 2 above)
Comment (rileynewton) [reply .github/workflows/snyk-scan.yml:7]: I believe I can change this to 
`on:
  push:
    branches: '**'
  pull_request_target:
      types: [opened, reopened, edited, synchronize]`
 
 which probably is preferable.

Comment (rileynewton) [reply .github/workflows/snyk-scan.yml]: Works, changed 🚀

**Pull request merged**